src/plugins/intel_cpu/src/nodes/segment_max.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "segment_max.h"
+
+#include "openvino/op/segment_max.hpp"
+#include "openvino/reference/segment_max.hpp"
+
+namespace ov::intel_cpu::node {
+SegmentMax::SegmentMax(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW_NOT_IMPLEMENTED(errorMessage);
+    }
+    fillMode = ov::as_type_ptr<const ov::op::v16::SegmentMax>(op)->get_fill_mode();
+}
+
+bool SegmentMax::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (!ov::is_type<ov::op::v16::SegmentMax>(op)) {
+            errorMessage = "Only opset16 SegmentMax operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void SegmentMax::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset16::SegmentMax
+}
+
+void SegmentMax::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty()) {
+        return;
+    }
+    ov::element::Type dataPrecision = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type segmentIdsPrecision = getOriginalInputPrecisionAtPort(1);
+    if (getOriginalInputsNumber() == 2) {
+        addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision}, {LayoutType::ncsp, segmentIdsPrecision}},
+                             {{LayoutType::ncsp, dataPrecision}},
+                             impl_desc_type::ref);
+    } else {
+        ov::element::Type numSegmentsPrecision = getOriginalInputPrecisionAtPort(2);
+        addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision},
+                              {LayoutType::ncsp, segmentIdsPrecision},
+                              {LayoutType::ncsp, numSegmentsPrecision}},
+                             {{LayoutType::ncsp, dataPrecision}},
+                             impl_desc_type::ref);
+    }
+}
```

Comment:
In the previous comments we have discussed the auto conversion of i64 to i32, in some ops it is forced by setting i32 as  addSupportedPrimDesc. It could be an option here, but if the tests doesn't complain, maybe it's not needed.

Some examples:
https://github.com/openvinotoolkit/openvino/blob/8ab7374e3de12dff0986191aadad1dca996a4610/src/plugins/intel_cpu/src/nodes/adaptive_pooling.cpp#L118-L123 
https://github.com/openvinotoolkit/openvino/blob/8ab7374e3de12dff0986191aadad1dca996a4610/src/plugins/intel_cpu/src/nodes/col2im.cpp#L45-L47

Reply:
There's no difference in actual secondary input types when hardcoding i32 vs parametrizing it, but I hardcoded i32 explicitly anyway for clarity.